lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSIdentityStorageImpl.java

```diff
@@ -32,35 +69,761 @@
  * Oct 5, 2015  
  */
 public class RDBMSIdentityStorageImpl extends IdentityStorageImpl {
+
+  private ActivityDAO activityDAO = null;
+  private IdentityDAO identityDAO = null;
+  private ProfileDAO profileDAO = null;
+  private SpaceStorage spaceStorage = null;
+
+  private final DataInitializer dataInitializer;
+
+  private ProfileSearchConnector profileSearchConnector;
+
+  public RDBMSIdentityStorageImpl(DataInitializer dataInitializer) {
+    System.out.println("Initial here");
+    this.dataInitializer = dataInitializer;
+  }
+
+  private ActivityDAO getActivityDAO() {
+    if (activityDAO == null) {
+      activityDAO = CommonsUtils.getService(ActivityDAO.class);
+    }
+    return activityDAO;
+  }
+
+  private IdentityDAO getIdentityDAO() {
+    if (identityDAO == null) {
+      identityDAO = CommonsUtils.getService(IdentityDAO.class);
+
+      //TODO: This is temporary to workaround, please consider
```

Comment:
This is workaround for common issue https://jira.exoplatform.org/browse/COMMONS-478. I updated the comment in code for more clear.